### PR TITLE
Refactor/andes color

### PR DESCRIPTION
**Pull request by carrymaster**
## Motivation
The colors are a mess cause we have frictions between ColorRes and ColorRaw. So we create a AndesColor to wrap this and not dependes on Android primitives types.

## Affected component
AndesMessage

## Checks
Are you sure you followed all those steps? In such case, let's say with me "I solemnly declare that ...":
- [x] I have coded an example inside the Demo App,
- [ ] I have added proper tests,
- [x] I have modified the [Changelog](https://github.com/mercadolibre/...) file,
- [ ] I have updated GitHub wiki,
- [ ] I have the explicit approval of one or more members of the UX Team,
- [x] I have updated the version in gradle.properties file.

## Change type
This is easy, let us know what this code is about:
- [ ] This code adds a new component
- [x] This code includes improvements to an existent component
- [ ] This code improves or modifies ONLY the Demo App.

### CODE REVIEW

**Comment by furyGorgoryAdmin** [comment]
### Informative Menu <h2>  ### 

### Use any of these tags in a pull request description or leave a comment ### 
| Tag | Description |
|---|---|
| **[canary]**| Creates a build that can only be deployed with canary strategy. | 
| **[build]** |  Creates version or build from a branch that has been tagged as "no releasable". | 
| **[name:version-name]** |Specify the name of the build. For example: [build] [name:1.2.3-name] or [canary] [name:1.2.3-name].|
| **[major]** | To automatically increase the first position of the version. For example: [build] [major] or [canary] [major].|
| **[minor]** | To automatically increase the second position of the version. For example: [build] [minor] or [canary] [minor]. |
| **[patch]** | To automatically increase the third position of the version. For example: [build] [patch] or [canary] [patch].|  

[For more information about ci-cd](http://docs.furycloud.io/?lang=esp&section=businessassurance)
[For more information about semver](https://semver.org/#summary)

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,11 +1,11 @@
-# v1.2.3
-# Added
-- Stale Bot GitHub config.
-- Buttons can have primary and secondary actions.
-
 # v1.2.2
-# Fixed
+## Fixed
```
Comment (lautimfernandez) [inline CHANGELOG.md]: Segun esta en el resto del changelog esta siempre con un solo '#'

File: components/src/main/java/com/mercadolibre/android/andesui/message/factory/AndesMessageConfigurationFactory.kt [hunk 2]
```diff
@@ -51,31 +52,32 @@ internal object AndesMessageConfigurationFactory {
...
-    private fun resolvePrimaryActionTextColor(hierarchy: AndesMessageHierarchyInterface, context: Context) = hierarchy.primaryActionTextColor(context)
-    private fun resolveSecondaryActionBackgroundColor(hierarchy: AndesMessageHierarchyInterface, type: AndesMessageTypeInterface, context: Context) = hierarchy.secondaryActionBackgroundColor(context, type)
-    private fun resolveSecondaryActionTextColor(hierarchy: AndesMessageHierarchyInterface, type: AndesMessageTypeInterface, context: Context) = hierarchy.secondaryActionTextColor(context, type)
+    private fun resolveDismissableIconColor(hierarchy: AndesMessageHierarchyInterface) = hierarchy.dismissableIconColor()
+    private fun resolvePrimaryActionBackgroundColor(hierarchy: AndesMessageHierarchyInterface, state: AndesMessageTypeInterface, context: Context) = hierarchy.primaryActionBackgroundColor(state)
+    private fun resolvePrimaryActionTextColor(hierarchy: AndesMessageHierarchyInterface) = hierarchy.primaryActionTextColor()
+    private fun resolveSecondaryActionBackgroundColor(hierarchy: AndesMessageHierarchyInterface, state: AndesMessageTypeInterface, context: Context) = hierarchy.secondaryActionBackgroundColor(state)
```
Comment (lautimfernandez) [inline components/src/main/java/com/mercadolibre/android/andesui/message/factory/AndesMessageConfigurationFactory.kt]: Creo que te mergeaste con los cambios de state, fijate que sea `type: AndesMessageTypeInterface` como estaba antes

File: components/src/main/java/com/mercadolibre/android/andesui/message/hierarchy/AndesMessageHierarchyInterface.kt [hunk 3]
```diff
@@ -89,33 +85,52 @@ internal sealed class AndesMessageHierarchyInterface {
...
 }
 
 internal object AndesLoudMessageHierarchy : AndesMessageHierarchyInterface() {
-    override fun backgroundColor(context: Context) = throw IllegalStateException("Loud message cannot be colored without an AndesMessageType")
-    override fun backgroundColor(context: Context, type: AndesMessageTypeInterface) = type.primaryColor(context)
-    override fun textColor(context: Context) = ContextCompat.getColor(context, R.color.andesui_message_loud_text)
-    override fun dismissableIconColor(context: Context) = ContextCompat.getColor(context, R.color.andesui_message_loud_dismissable)
-    override fun iconBackgroundColor(context: Context, type: AndesMessageTypeInterface) = type.secondaryColor(context)
-    override fun secondaryActionTextColor(context: Context, type: AndesMessageTypeInterface): Int = ContextCompat.getColor(context, R.color.andesui_white)
+    override fun backgroundColor() = throw IllegalStateException("Loud message cannot be colored without an AndesMessageState")
```
Comment (lautimfernandez) [inline components/src/main/java/com/mercadolibre/android/andesui/message/hierarchy/AndesMessageHierarchyInterface.kt]: Mismo caso, quedo como State y tiene que ser Type

**Pull request merged**